improver/cli/apply_samos_coefficients.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Script to apply Standardised Anomaly Model Output
+Statistics (SAMOS) calibration."""
+
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    *cubes: cli.inputcubelist,
+    gams: cli.inputpickle,
+    gam_features: cli.comma_separated_list,
+    validity_times: cli.comma_separated_list = None,
+    realizations_count: int = None,
+    randomise=False,
+    random_seed: int = None,
+    ignore_ecc_bounds_exceedance=False,
+    tolerate_time_mismatch=False,
+    predictor="mean",
+    percentiles: cli.comma_separated_list = None,
+):
```

Comment:
The underlying ApplySAMOS plugin will be modified to require gams for both the forecast and truths to be provided to the plugin. This will also be required for this CLI.

```suggestion
@cli.clizefy
@cli.with_output
def process(
    *cubes: cli.inputcubelist,
    forecast_gams: cli.inputpickle,
    truth_gams: cli.inputpickle,
    gam_features: cli.comma_separated_list,
    validity_times: cli.comma_separated_list = None,
    realizations_count: int = None,
    randomise=False,
    random_seed: int = None,
    ignore_ecc_bounds_exceedance=False,
    tolerate_time_mismatch=False,
    predictor="mean",
    percentiles: cli.comma_separated_list = None,
):
```

Reply:
My comment above should be ignored - the forecast and truth GAMs will be passed around in a single pickle file (see [here](https://github.com/metoppv/mo-blue-team/issues/948#issuecomment-3204796950)).

Reply:
It's a rollercoaster.